File: AppClient/src/app/model/application.ts
export type ApplicationStatus =
  | 'DRAFT'
  | 'SENT'
  | 'INTERVIEW_SCHEDULED'
  | 'INTERVIEW_DONE'
  | 'OFFER_RECEIVED'
  | 'ACCEPTED'
  | 'REJECTED'
  | 'WITHDRAWN';

export interface Application {
  id?: number;
  companyPositionId: number;
  companyName: string;
  positionTitle: string;
  status: ApplicationStatus;
  appliedDate?: string;
  notes?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface ApplicationRequest {
  companyPositionId: number;
  status: ApplicationStatus;
  appliedDate?: string | null;
  notes?: string | null;
}

File: src/main/java/de/samply/manager/model/Application.java
package de.samply.manager.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.time.LocalDate;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

@Data
@ToString
@EqualsAndHashCode
@Entity
@Table(name = "applications")
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String userId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_position_id", nullable = false)
    private CompanyPosition companyPosition;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ApplicationStatus status = ApplicationStatus.DRAFT;

    private LocalDate appliedDate;

    private String notes;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cover_letter_document_id")
    private Document coverLetter;

    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @CreationTimestamp
    private LocalDateTime updatedAt;
}
